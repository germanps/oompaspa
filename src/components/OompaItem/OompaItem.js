import React from 'react'
import './OompaItem.scss'
import { formatGender } from '../../Utils/utils'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export const OompaItem = ({ oompa }) => {
    return (

        <div className="oompa-item card">
            <img src={oompa.image} className="card-img-top" alt={oompa.first_name} />
            <div className="card-body oompa-item__body">
                <Link to={`./oompa/${oompa.id}`} className="oompa-item__body--link">
                    <h3 className="title">{oompa.first_name}</h3>
                </Link>
                <p className="oompa-item__body--text">{formatGender(oompa.gender)}</p>
                <p className="oompa-item__body--text">{oompa.profession}</p>
            </div>
        </div>

    )
}

OompaItem.propTypes = {
    oompa: PropTypes.object.isRequired
}