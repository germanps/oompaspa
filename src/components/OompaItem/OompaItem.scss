@import '../../scss/index.scss';

.oompa-item{
    width: 100%;
    border: none;
    margin-bottom: 1.5rem;
    &__body{
        padding: .5rem 0;

        &--link{
            color: $font-dark;
            transition: color .15s ease;
            .title{
                font-size: $font-size;
                font-weight: bold;
                margin-bottom: .3rem;
            }
            &:hover{
                text-decoration: none;
                color: $high-blue;
            }
        }
        &--text{
            margin-bottom: 0;
            font-size: $font-size / 1.2;
        }
    }
}