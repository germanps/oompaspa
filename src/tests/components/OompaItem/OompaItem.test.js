import React from 'react'
import { shallow } from 'enzyme'
import { OompaItem } from '../../../components/OompaItem/OompaItem'

describe('Test in <OompaItem />', () => {
    test('should show component correctly with its props', () => {
        const oompa = {
            id: 1,
            first_name: 'Test',
            image: 'some url or link',
            gender: 'M',
            profession: 'Something'
        }
        const wrapper = shallow(<OompaItem oompa={oompa} />)
        expect(wrapper).toMatchSnapshot();
    })

})
