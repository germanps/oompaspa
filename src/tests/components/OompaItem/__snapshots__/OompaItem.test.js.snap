// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <OompaItem /> should show component correctly with its props 1`] = `
<div
  className="oompa-item card"
>
  <img
    alt="Test"
    className="card-img-top"
    src="some url or link"
  />
  <div
    className="card-body oompa-item__body"
  >
    <Link
      className="oompa-item__body--link"
      to="./oompa/1"
    >
      <h3
        className="title"
      >
        Test
      </h3>
    </Link>
    <p
      className="oompa-item__body--text"
    >
      Man
    </p>
    <p
      className="oompa-item__body--text"
    >
      Something
    </p>
  </div>
</div>
`;
